src/core.c/io_operators.rakumod

```diff
@@ -282,34 +282,77 @@ multi sub indir(IO() $path, &what, :$d = True, :$r, :$w, :$x) {
     }
 }
 
-proto sub chmod($, |) {*}
-multi sub chmod($mode, *@filenames) {
+proto sub chmod($, |) is revision-gated("6.c") {*}
+multi sub chmod($mode, Str $path) is revision-gated("6.c") { $path.IO.chmod($mode) }
+multi sub chmod($mode, IO::Path $path) is revision-gated("6.c") { $path.chmod($mode) }
+# In 6.e, we adjust the routine arguments to align with how chown is called
+# So we set up 6.e variants of the old signature that include deprecation messages
+# We also provide the new signature and make it available to 6.e+
+multi sub chmod($mode, Str $path) is revision-gated("6.e") {
+    DEPRECATED('chmod(Str $path, :$mode)', :what("old-fashioned chmod signature"));
+    $path.IO.chmod($mode)
+}
+multi sub chmod($mode, IO::Path $path) is revision-gated("6.e") {
+    DEPRECATED('chmod(IO::Path $path, :$mode)', :what("old-fashioned chmod signature"));
+    $path.chmod($mode)
+}
+multi sub chmod(Str $path, :$mode!) is revision-gated("6.e") { $path.IO.chmod($mode) }
```

Comment:
The `Str` and `IO::Path` typing for non 6.e candidates is technically a breaking change and we probably shouldn't change that.